idflakies-maven-plugin/src/main/java/edu/illinois/cs/dt/tools/plugin/IncDetectorMojo.java

```diff
@@ -0,0 +1,635 @@
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static edu.illinois.starts.helpers.ZLCHelper.STAR_FILE;
+
+@Mojo(name = "incdetect", defaultPhase = LifecyclePhase.TEST, requiresDependencyResolution = ResolutionScope.TEST)
+public class IncDetectorMojo extends DetectorMojo {
+
+    protected static String CLASSES = "classes";
+    protected static String EQUAL = "=";
+    protected static String JAR_CHECKSUMS = "jar-checksums";
+    protected static String SF_CLASSPATH = "sf-classpath";
+    protected static String TEST_CLASSES = "test-classes";
+    private static final String TARGET = "target";
+
+    /**
+     * The directory in which to store STARTS artifacts that are needed between runs.
+     */
+    protected String artifactsDir;
+
+    protected String startsDir;
+
+    protected String ekstaziDir;
+
+    protected ClassLoader loader;
+
+    protected List<Pair> jarCheckSums = null;
+
+    protected Set<String> selectedTests;
+
+    // the dependency map from test classes to their dependencies (classes)
+    protected Map<String, Set<String>> transitiveClosure;
+
+    // the dependency map from classes to their dependencies (test classes)
+    protected Map<String, Set<String>> reverseTransitiveClosure;
+
+    private Classpath sureFireClassPath;
+
+    protected boolean selectMore;
+
+    protected boolean detectOrNot;
+
+    protected boolean selectAll;
+
+    protected Path ekstaziSelectedTestsPath;
+
+    protected String ekstaziDependenciesFile;
+
+    protected Path startsSelectedTestsPath;
+
+    protected Path startsDependenciesPath;
+
+    protected boolean isEkstazi;
+
+    private Set<String> affectedTestClasses;
+
+    private static Set<String> immutableList;
+
+    @Override
+    public void execute() {
+        superExecute();
+
+        final ErrorLogger logger = new ErrorLogger();
+        this.outputPath = PathManager.detectionResults();
+        this.coordinates = mavenProject.getGroupId() + ":" + mavenProject.getArtifactId() + ":" + mavenProject.getVersion();
+
+        logger.runAndLogError(() -> defineSettings(logger, mavenProject));
+        if (this.runner == null) {
+            return;
+        }
+        long startTime = System.currentTimeMillis();
+        try {
+            affectedTestClasses = computeAffectedTests(mavenProject);
+            if (!detectOrNot) {
```

Comment:
Since if this is false then can just return, should it be checked before doing all the work with computing affected tests?